app/models/cashbox.rb

```diff
@@ -0,0 +1,7 @@
+class Cashbox < ActiveRecord::Base
+  has_many :deposits, class_name: 'Payment', foreign_key: 'destination_id'
+  has_many :withdrawls, class_name: 'Payment', foreign_key: 'source_id'
```

Comment:
@Maroo-b Yes, I see.  I like the way you did it, aside from disagreeing with storing a balance and the Payment model isn't updated with its `belongs_to` associations